pkg/discovery/network/calico.go

```diff
@@ -0,0 +1,47 @@
+/*
+© 2021 Red Hat, Inc. and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package network
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func discoverCalicoNetwork(clientSet kubernetes.Interface) (*ClusterNetwork, error) {
+	_, err := clientSet.CoreV1().ConfigMaps("kube-system").Get(context.TODO(), "calico-config", metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, errors.WithMessage(err, "error obtaining the \"calico\" ConfigMap")
+	}
+
+	clusterNetwork, err := discoverNetwork(clientSet)
+	if err != nil {
+		return nil, err
+	}
+
+	if clusterNetwork != nil {
+		clusterNetwork.NetworkPlugin = "calico"
```

Comment:
The networkPlugin names should be made as Constants and the same should be referred from different repos. 
We need changes even in route-agent to use those constants otherwise it will cause regression 

https://github.com/submariner-io/submariner/blob/ca6d02e1efe507c209fd2d1eba2b301937daea94/pkg/routeagent_driver/handlers/kubeproxy/kp_iptables.go#L82

Reply:
@sridhargaddam - see PR here:
https://github.com/submariner-io/submariner/pull/1354